src/Microsoft.ML.KMeansClustering/KMeansPlusPlusTrainer.cs

```diff
@@ -45,34 +45,54 @@ public enum InitAlgorithm
         }
 
         [BestFriend]
-        internal static class Defaults{
+        internal static class Defaults
+        {
             /// <value>The number of clusters.</value>
-            public const int K = 5;
+            public const int ClustersCount = 5;
         }
 
-        public class Arguments : UnsupervisedLearnerInputBaseWithWeight
+        public class Options : UnsupervisedLearnerInputBaseWithWeight
         {
-            [Argument(ArgumentType.AtMostOnce, HelpText = "The number of clusters", SortOrder = 50)]
+            /// <summary>
+            /// The number of clusters.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce, HelpText = "The number of clusters", SortOrder = 50, Name = "K")]
```

Comment:
>Name [](start = 100, length = 4)

ShortName maybe? it's a "K" is quite short, right?

Reply:
Name is for backwards compatibility. This is what maml uses. 

---
In reply to: [248405583](https://github.com/dotnet/machinelearning/pull/2158#discussion_r248405583) [](ancestors = 248405583)

Reply:
Apparently name is for maml, and i set it to the previous name for backwards compatibility. 

---
In reply to: [248405874](https://github.com/dotnet/machinelearning/pull/2158#discussion_r248405874) [](ancestors = 248405874,248405583)

Reply:
ShortNames for same thing, and as @najeeb-kazmi discover yesterday, Name is not always work, but ShortName do (or maybe combination of ShortName + Name).

---
In reply to: [248405874](https://github.com/dotnet/machinelearning/pull/2158#discussion_r248405874) [](ancestors = 248405874,248405583)